### ISSUE TITLE
ZStream: add intersperse

### ISSUE THREAD

**Issue by mvillafuertem** [zio/zio#3440]
Hi, maybe this option is interesting for users

This is present in Akka Streams

https://doc.akka.io/docs/akka/current/stream/operators/Source-or-Flow/intersperse.html

```
Source(1 to 4).map(_.toString).intersperse("[", ", ", "]").runWith(Sink.foreach(print))
```

It's great when you have a streaming api, and always you would send a valid json

```
Source(jsonTweets).intersperse("[", ", ", "]").runWith(Sink.foreach(print))
```

Thanks!

**Comment by fagossa** [comment]
I would like to take this one

**Comment by fagossa** [comment]
@iravid I got a WIP PR in here https://github.com/zio/zio/pull/3514 . I still got some issues because I'm quite new to the base code.

**Comment by iravid** [comment]
Closed in #3514
